cirq-core/cirq/transformers/gauge_compiling/sqrt_cz_gauge.py

```diff
@@ -14,16 +14,44 @@
 
 """A Gauge transformer for CZ**0.5 gate."""
 
+import numpy as np
+
 from cirq.transformers.gauge_compiling.gauge_compiling import (
     GaugeTransformer,
     GaugeSelector,
     ConstantGauge,
+    Gauge,
 )
 from cirq.ops.common_gates import CZ
 from cirq import ops
 
-SqrtCZGaugeSelector = GaugeSelector(
-    gauges=[ConstantGauge(pre_q0=ops.X, post_q0=ops.X, post_q1=ops.Z**0.5, two_qubit_gate=CZ**-0.5)]
-)
 
-SqrtCZGaugeTransformer = GaugeTransformer(target=CZ**0.5, gauge_selector=SqrtCZGaugeSelector)
+class SqrtCZGauge(Gauge):
+
+    def weight(self) -> float:
+        return 5.0
+
+    def sample(self, gate: ops.Gate, prng: np.random.Generator) -> ConstantGauge:
+        if prng.choice([True, False]):
+            return ConstantGauge(two_qubit_gate=gate)
+        if prng.choice([True, False]):
+            return ConstantGauge(
+                pre_q0=ops.X,
+                post_q0=ops.X,
+                post_q1=ops.S if gate == CZ**0.5 else ops.S**-1,
+                two_qubit_gate=gate**-1,
+                swap_qubits=prng.choice([True, False]),
```

Comment:
Are you sure this is necessary?  `CZ ** +0.5` and `CZ ** -0.5` are both symmetric for the exchange of `q0 <--> q1`; the resulting operations are the same.  Unless you'll need `swap_qubits` for other gauge transformers in the future, it does not seem necessary in ConstantGauge at this time.

Reply:
fixed